src/lib/coda_commands/coda_commands.ml

```diff
@@ -172,14 +113,52 @@ let reset_trust_status t (ip_address : Unix.Inet_addr.Blocking_sexp.t) =
   let trust_system = config.trust_system in
   Trust_system.reset trust_system ip_address
 
-let setup_user_command ~fee ~nonce ~valid_until ~memo ~sender_kp
-    user_command_body =
-  let payload =
-    User_command.Payload.create ~fee ~nonce ~valid_until ~memo
-      ~body:user_command_body
+let replace_block_production_keys keys pks =
+  let kps =
+    List.filter_map pks ~f:(fun pk ->
+        let open Option.Let_syntax in
+        let%map kps =
+          Coda_lib.wallets keys |> Secrets.Wallets.find_unlocked ~needle:pk
+        in
+        (kps, pk) )
+  in
+  Coda_lib.replace_block_production_keypairs keys
+    (Keypair.And_compressed_pk.Set.of_list kps) ;
+  kps |> List.map ~f:snd
+
+let setup_and_submit_user_command t (user_command_input : User_command_input.t)
+    =
+  let open Participating_state.Let_syntax in
+  let%map account_opt = get_account t user_command_input.sender in
+  let open Deferred.Let_syntax in
+  let%map result = Coda_lib.add_transactions t [user_command_input] in
+  txn_count := !txn_count + 1 ;
+  match result with
+  | Ok ([], [failed_txn]) ->
+      Error
+        (Error.of_string
+           (sprintf !"%s"
+              ( Network_pool.Transaction_pool.Resource_pool.Diff.Diff_error
+                .to_yojson (snd failed_txn)
+              |> Yojson.Safe.to_string )))
+  | Ok ([txn], []) ->
+      Ok (txn, record_payment t txn (Option.value_exn account_opt))
+  | Ok _ ->
+      Error (Error.of_string "Invalid result from scheduling a payment")
+  | Error e ->
+      Error e
+
+let setup_and_submit_user_commands t user_command_list =
+  let open Participating_state.Let_syntax in
+  let%map _is_active = Coda_lib.active_or_bootstrapping t in
+  let logger =
+    Logger.extend
```

Comment:
Since you extending this logger and only using it once, you could just simplify this code by passing the original logger into the log function and adding the metadata to that call using the `~metadata` optional argument.